src/parsec_client/wire_protocol.md

```diff
@@ -218,86 +225,100 @@ messages on any suitable transport medium.
+   remain the same across different wire protocol versions.
+- Header size (_common_): a 16-bit unsigned integer whose value is the size of the **remainder** of
+   the header in bytes (once the magic number and header size fields have been consumed).
+   Consumers **must** use this field to consume the correct number of bytes of header from the
+   input stream, rather than use this specification to deduce the header size. This field's 
+   position and width will remain the same across different wire protocol versions. Only the value 
+   of this field may change between versions.
+- Major version number (_common_): an 8-bit versioning field. Currently the only supported and 
+   valid value for this field is 1. This field's position and width will remain the same across 
+   different wire protocol versions. Only the value of this field may change between versions.
+- Minor version number (_common_): an 8-bit versioning sub-field. Currently the only supported and 
+   valid value for this field is 0. This field's position and width will remain the same across 
+   different wire protocol versions. Only the value of this field may change between versions.
+- Flags (_common_): a 16-bit field that is currently unused and should be set to zero.
+- Provider (_common_): an 8-bit field identifying the back-end service provider for which the 
+   request is intended. A value of zero indicates that the request is intended for a special 
+   provider, which always exists, and is used for service discovery and communication bootstrapping.
+- Session handle (_common_): an 8-byte session identifier.
+- Content type (_common_): an 8-bit field that defines how the request body should be processed. 
+   The only currently-supported value is 1, which indicates that the request body should be treated 
+   as a serialized protobuf message.
+- Accept type (_requests only_): an 8-bit field that defines how the service should provide its 
+   response. The only currently-supported value is 1, which indicates that the service should 
+   provide a response whose body is a serialized protobuf message.
+- Auth type (_requests only_): an 8-bit field that defines how the authentication bytes should be 
+   interpreted.
+- Content length (_common_): a 32-bit unsigned integer field providing the exact number of bytes of 
+   content.
+- Auth length (_requests only_): a 16-bit unsigned integer field providing the exact number of 
+   bytes of authentication.
+- OPCODE (_common_): the 16-bit unsigned integer opcode, indicating the operation being performed 
+   by this request. See the section above on opcodes.
+- STATUS (_responses only_): the 16-bit unsigned integer status, indicating the overall success or 
+   failure of the operation. A value of zero is used universally to mean success. Other values 
+   should be interpreted according to the API specification.
+- RESERVED (_common_): a 16-bit field that is currently unused and must be set to zero.
```

Comment:
In the interface we have `reserved1` and `reserved2` both 8-bit long. Any reason why not merging them in the interface as well?

Reply:
No significant reason. As long as it's a 2-byte space at this stage it doesn't matter too much. We could merge them in the interface for consistency, but realistically nobody should even be looking at or disseminating them. They may well never be used.